### ISSUE TITLE
UseShortArrayInitializer with arrays of parametrized types (generic array creation)

### ISSUE THREAD

**Issue by harbulot** [pmd/pmd#4576]
**Affects PMD Version:** 7.0.0-rc2

**Rule:** `UseShortArrayInitializer`

https://docs.pmd-code.org/pmd-doc-7.0.0-rc2/pmd_rules_java_codestyle.html#useshortarrayinitializer

**Code Sample demonstrating the issue:**

```java
Set<Integer> set1 = new HashSet<>();

@SuppressWarnings("unchecked")		
Set<Integer>[] setArray = new Set[] { set1 };
```

**Expected outcome:**

The code above compiles correctly, but PMD reports a `UseShortArrayInitializer` rule violation on the last statement.

This effectively suggests to use this instead:

```java
Set<Integer>[] setArray = { set1 };
```

Unfortunately, this results in a "Compilation failure" with "generic array creation".

Using the longer form with `new Set[] { ... }` appears to be necessary when it's an array of parametrized type.

**Running PMD through:** Maven (using maven-pmd-plugin version 3.20.1-pmd-7-SNAPSHOT configured with PMD 7.0.0-rc2)
